stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/DatabaseChannelRepository.kt

```diff
@@ -258,11 +210,23 @@ internal class DatabaseChannelRepository(
                 channel.apply {
                     lastMessageAt = messageCreatedAt
                     messages = listOf(lastMessage)
-                }.also { insertChannel(it) }
+                }.also { upsertChannel(it) }
             }
         }
     }
 
+    private suspend fun Channel.parseEntity(): ChannelEntity {
+        val dbChannel = channelDao.select(this.cid)
+
+        return if (dbChannel?.lastMessageAt?.after(this.createdAt) == true) {
```

Comment:
I guess you wanted to compare message dates, right?
```suggestion
        return if (dbChannel?.lastMessageAt?.after(this.lastMessageAt) == true) {
```